siege_utilities/config/clients.py

```
@@ -4,14 +4,32 @@
+# alphanumeric+underscore+dash form so callers can't smuggle path
+# components like "../etc" into client_<code>.json.
+_CLIENT_CODE_RE = re.compile(r"^[A-Za-z0-9_\-]{1,64}$")
+
+
+def _validate_client_code(code: str) -> str:
+    if not isinstance(code, str) or not _CLIENT_CODE_RE.match(code):
+        raise ValueError(
+            f"client_code {code!r} must match [A-Za-z0-9_-]{{1,64}} — "
+            "it becomes part of a filename, so path separators, dots, "
+            "and shell metacharacters are rejected."
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)